cmd/multiclusterapp.go

```diff
@@ -0,0 +1,703 @@
+		Usage:   "Operations with multi-cluster apps",
+		Action:  defaultAction(multiClusterAppLs),
+		Flags:   appLsFlags,
+		Subcommands: []cli.Command{
+			cli.Command{
+				Name:        "ls",
+				Usage:       "List multi-cluster apps",
+				Description: "\nList all multi-cluster apps in the current Rancher server",
+				ArgsUsage:   "None",
+				Action:      multiClusterAppLs,
+				Flags:       appLsFlags,
+			},
+			cli.Command{
+				Name:      "delete",
+				Usage:     "Delete a multi-cluster app",
+				Action:    multiClusterAppDelete,
+				ArgsUsage: "[APP_NAME]",
+			},
+			cli.Command{
+				Name:        "install",
+				Usage:       "Install a multi-cluster app",
+				Description: installMultiClusterAppDescription,
+				Action:      multiClusterAppTemplateInstall,
+				ArgsUsage:   "[TEMPLATE_NAME, APP_NAME]...",
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name:  "answers,a",
+						Usage: "Path to an answers file, the format of the file is a map with key:value. This supports JSON and YAML.",
+					},
+					cli.StringFlag{
+						Name:  "values",
+						Usage: "Path to a helm values file.",
+					},
+					cli.StringSliceFlag{
+						Name:  "set",
+						Usage: "Set answers for the template, can be used multiple times. Example: --set foo=bar",
+					},
+					cli.StringFlag{
+						Name:  "version",
+						Usage: "Version of the template to use",
+					},
+					cli.BoolFlag{
+						Name:  "no-prompt",
+						Usage: "Suppress asking questions and use the default values when required answers are not provided",
+					},
+					cli.StringSliceFlag{
+						Name:  "target,t",
+						Usage: "Target project ids to install the app into",
+					},
+				},
+			},
+			cli.Command{
+				Name:      "rollback",
+				Usage:     "Rollback a multi-cluster app to a previous version",
+				Action:    multiClusterAppRollback,
+				ArgsUsage: "[APP_NAME/APP_ID, REVISION_ID/REVISION_NAME]",
+				Flags: []cli.Flag{
+					cli.BoolFlag{
+						Name:  "show-revisions,r",
+						Usage: "Show revisions available to rollback to",
+					},
+				},
+			},
+			cli.Command{
+				Name:      "upgrade",
+				Usage:     "Upgrade an app to a newer version",
+				Action:    multiClusterAppUpgrade,
+				ArgsUsage: "[APP_NAME/APP_ID VERSION]",
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name:  "answers,a",
+						Usage: "Path to an answers file, the format of the file is a map with key:value. Supports JSON and YAML",
+					},
+					cli.StringFlag{
+						Name:  "values",
+						Usage: "Path to a helm values file.",
+					},
+					cli.StringSliceFlag{
+						Name:  "set",
+						Usage: "Set answers for the template, can be used multiple times. Example: --set foo=bar",
+					},
+					cli.BoolFlag{
+						Name:  "show-versions,v",
+						Usage: "Display versions available to upgrade to",
+					},
+					cli.StringSliceFlag{
+						Name:  "target,t",
+						Usage: "Target project ids to upgrade, leave it empty to keep current targets",
```

Comment:
This comment needs some clarity, if you drop a target on upgrade it's gone, it doesn't mean it's running the old version still. If I have target1 and target2 then say `upgrade --target target1` target2 gets dropped completely.